File: c++/Read_bit.cpp
#include "Read_bit.h"

Read_bit::Read_bit(const std::string &fileName){
    read_stream.open(fileName);
    if(!read_stream.is_open()){
        std::cerr << "No file found!" << std::endl;
    }
    buffer=0;
    bits_in_buffer=0;
}

void Read_bit::operator>>(bool &bit){
    bit=0;
    if(!read_stream.eof()){
        if(bits_in_buffer==0){
            c=read_stream.get();
            bits_in_buffer = 8;
        }
        bit = c >> (CHAR_BIT-1);
        bits_in_buffer--;
        c<<=1;
        c&=0xFF;
    }
    return;
}

File: python/Read_bit.py
class Read_bit:
    def __init__(self,namefile):
        self.namefile = namefile
        self.file = open(namefile,"rb")
        self.buffer=0;
        self.bits_in_buffer=0;
        self.CHAR_BIT = 8

    def read(self):
        bit=0
        if(self.bits_in_buffer==0):
            try:
                self.char = ord(self.file.read(1))
                self.bits_in_buffer = self.CHAR_BIT
            except:
                self.char = 0
        bit = self.char >> (self.CHAR_BIT-1)
        self.bits_in_buffer-=1
        self.char<<=1
        self.char&=0xFF
        return bit

    def close(self):
        self.file.close();
        return



